portfolio/src/main/java/com/google/sps/servlets/CreatePuzzleServlet.java

```diff
@@ -64,14 +64,18 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
     Puzzle puzzle = new Puzzle();
 
     puzzle.setImageUrl(uploadedFileUrl);
+    puzzle.setDifficulty(difficulty);
 
     PuzzleDao dao = new PuzzleDao();
 
     //Need to double check that puzzle is passed by reference
     dao.create(puzzle);
 
+    Long puzzleId = puzzle.getPuzzleId();
+    String redirectURL = "/ViewPuzzle?puzzleid=%s" + String.valueOf(puzzleId);
```

Comment:
```suggestion
    String redirectURL = "/ViewPuzzlePage.html?id=%s" + String.valueOf(puzzleId);
```
_ViewPuzzleServlet_ handles requests like `/puzzle?id=SOMEID` by returning json, the user should be redirected to `/ViewPuzzlePage.html?id=SOMEID`. 

 _ViewPuzzlePage.html_ automatically runs javascript that makes request to _ViewPuzzleServlet_.